pattoo_shared/daemon.py

```diff
@@ -277,3 +267,124 @@ def run(self):
         """
         # Simple comment to pass linter
         pass
+
+
+class GracefulDaemon(Daemon):
+    """Daemon that allows for graceful shutdown
+
+    This daemon should allow for stop/restart commands to perform graceful
+    shutdown of a given process. A graceful shutdown involves checking that
+    whether a current process is running and only ending the process once the
+    current process has completed its currently running task.
+
+    """
+    def __init__(self, agent):
+        """Initialize the class.
+
+        Args:
+            agent: Agent object
+
+        Returns:
+            None
+
+        """
+        try:
+            self.graceful_timeout = agent.config.graceful_timeout()
+        except AttributeError as err:
+            # Sets default GracefulDaemon shutdown timeout if not defined by the
+            # agent configuration
+            log_message = 'Graceful Timeout configuration not set, {}\n Default setting to 10s'.format(err)
+            log.log2info(1100, log_message)
+
+            self.graceful_timeout = 10
+
+        Daemon.__init__(self, agent)
+
+    def __daemon_running(self):
+        """Determines if daemon is running
+
+        Daemon is running based on whether it has an associated lockfile
+
+        Args:
+            None
+
+        Return:
+            running: True if daemon is currently running or conducing a process
+
+        """
+        running = False
+        if self.lockfile is not None:
+            if os.path.exists(self.lockfile) is True:
+                running = True
+
+        return running
+
+    def graceful_shutdown(self, fn):
```

Comment:
Use at least a three character variable name. This is a requirement for pylint. Add the pylint plugin to your IDE. We need to keep the look and feel consistent.